### PR TITLE
[gitpod-db] Don't count stopping instances as running

### LINKED ISSUE #4906: Treat stopping workspaces as not-running
### Bug description

When I have some workspaces in `Stopping` phase. They are still being counted as running and I get the error message that I have reached my limit of running parallel workspaces.

### Steps to reproduce

Start X workspaces matching your accounts parallel ws limit.
Stop them all.
Try starting a new ws when the older ones are still in stopping phase.

### Expected behavior

New workspaces should be started.
Stopping ws should not be counted towards the limit.

### Example repository

https://github.com/gitpod-io/gitpod

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by csweichel**
With this PR we no longer count `stopping` workspaces as running.

I've gone through the callers of `findRegularRunningInstances` and `findRunningInstancesWithWorkspaces`, and have not found a place where that change would adversely affect the behaviour with one minor exception:

In tthe cluster service the check if the cluster is empty using `findRegularRunningInstances`. Prior we would not consider the cluster empty if there were still stopping instances, now `stopping` instances no longer prevent cluster de-registration. This does not mean the workspaces would loose their backup, but the data in the database must become inconsistent. We should **never** even deregister a non-empty cluster to begin with. All things considered, I figured the additional variance/complexity of introducing something like `includeStoppingWorkspaces` just for this one case was unjustified.

fixes #4906

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4910#" title="Author self-approved">csweichel</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jankoehnlein** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jankoehnlein` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by svenefftinge — APPROVED** [review]

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: d1e7edccaa1b007cc24126e9ec9095280d17c1cf</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4910#" title="Author self-approved">csweichel</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/4910#pullrequestreview-712578524" title="Approved">svenefftinge</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [svenefftinge]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
